app-backend/tool/src/main/scala/ast/MapAlgebraAST.scala

```diff
@@ -9,9 +9,22 @@ import io.circe.generic.JsonCodec
 sealed trait MapAlgebraAST extends Product with Serializable {
   def id: UUID
   def args: List[MapAlgebraAST]
-  def label: Option[String]
+  def metadata: Option[NodeMetadata]
   def find(id: UUID): Option[MapAlgebraAST]
   def sources: Seq[UUID]
+
+  def overrideMetadata(otherMD: Option[NodeMetadata]): Option[NodeMetadata] =
+    (metadata, otherMD) match {
+      case (Some(defaults), Some(overrides)) =>
+        Some(NodeMetadata(
+          overrides.label.orElse(defaults.label),
+          overrides.description.orElse(defaults.description),
+          overrides.histogram.orElse(defaults.histogram)
+        ))
+      case (None, Some(_)) => otherMD
+      case (Some(_), None) => metadata
+      case _ => None
+    }
```

Comment:
I'm not entirely sure I know what this would mean/look like such that it remains readable